### PR TITLE
Muse connection

### PR DESCRIPTION
**Pull request by capitancambio**
First approach to link muse with the app

### CODE REVIEW

**Comment by monchote** [comment]
Can you squash your changes? It's a nightmare to review otherwise.

**Comment by capitancambio** [comment]
I've forced it!

**Comment by monchote** [comment]
WUT?

**Comment by capitancambio** [comment]
it's been squashed ...

**Comment by monchote** [comment]
It still shows as 8 commits.

**Comment by capitancambio** [comment]
You want just one?

File: .idea/misc.xml [hunk 1]
```diff
@@ -37,7 +37,7 @@
     <ConfirmationsSetting value="0" id="Add" />
     <ConfirmationsSetting value="0" id="Remove" />
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" default="true" assert-keyword="true" jdk-15="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="true" assert-keyword="true" jdk-15="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
```
Comment (monchote) [inline .idea/misc.xml]: Why the change?

File: .idea/modules.xml [hunk 2]
```diff
@@ -3,7 +3,9 @@
   <component name="ProjectModuleManager">
     <modules>
       <module fileurl="file://$PROJECT_DIR$/app/app.iml" filepath="$PROJECT_DIR$/app/app.iml" />
+      <module fileurl="file://$PROJECT_DIR$/app/app.iml" filepath="$PROJECT_DIR$/app/app.iml" />
```
Comment (monchote) [inline .idea/modules.xml]: These 2 lines seem to be duplicated.

File: app/src/main/java/uk/co/neuralcubes/neuralates/muse/MuseHandler.java [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+ *     <li> uk.co.neuralcubes.MuseHandler.AccelerometerReading </li>
+ *     <li> uk.co.neuralcubes.MuseHandler.FocusLevel </li>
+ *     <li> uk.co.neuralcubes.MuseHandler.HorseshoeReading </li>
+ *  </ul>
+ *
+ *
+ */
+public class MuseHandler implements Closeable {
+
+
```
Comment (monchote) [inline app/src/main/java/uk/co/neuralcubes/neuralates/muse/MuseHandler.java:39]: Too many empty lines.

**Comment by monchote** [comment]
I think, given the nature of the project, it's not practical to code-review every single change or commit.

I'd suggest having a single pull-request per "feature" where all the changes are related to each other. Having a squashed commit would be ideal for the commit history although I've just found that Github shows a "squashed view" of all the changes.

File: app/src/main/java/uk/co/neuralcubes/neuralates/muse/MuseHandler.java [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+    class AccelerometerReading {
+
+        private double x,y,z;
+
+        public AccelerometerReading(double x, double y, double z) {
+            this.z = z;
+            this.y = y;
+            this.x = x;
+        }
+
```
Comment (monchote) [inline app/src/main/java/uk/co/neuralcubes/neuralates/muse/MuseHandler.java]: Empty lines.

File: app/src/main/java/uk/co/neuralcubes/neuralates/muse/MuseHandler.java [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+
+        double getX(){
+            return x;
+        }
```
Comment (monchote) [inline app/src/main/java/uk/co/neuralcubes/neuralates/muse/MuseHandler.java:52]: Let's be consistent with empty lines. I'd advocate for a single empty line.

File: app/src/main/java/uk/co/neuralcubes/neuralates/muse/MuseHandler.java [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+        double getY(){
+            return y;
+        }
+        double getZ(){
+            return z;
+        }
+    }
+
+    class BatteryReading{
```
Comment (monchote) [inline app/src/main/java/uk/co/neuralcubes/neuralates/muse/MuseHandler.java]: I personally don't like these classes. They don't do anything that a parameter/method name would.

File: .idea/modules.xml (hunk 2 above)
Comment (capitancambio) [inline .idea/modules.xml]: With this and the previous comment, I usually don't mess with the IDE config files, these changes have been automatic and I'd prefer let android studio do it's own thing

File: app/src/main/java/uk/co/neuralcubes/neuralates/muse/MuseHandler.java (hunk 6 above)
Comment (capitancambio) [inline app/src/main/java/uk/co/neuralcubes/neuralates/muse/MuseHandler.java]: They encapsulate values so we can send them down the EventBus, otherwise the subscribed observers wouldn't know which class expect. We could refactor the communication to Listener interface instead of an event bus

**Comment by monchote** [comment]
Ship it!
